kernel/h1/src/hil/flash/driver.rs

```diff
@@ -63,13 +67,36 @@ impl<'d, A: Alarm<'d>, H: Hardware> FlashImpl<'d, A, H> {
 }
 
 const MAX_WRITE_SIZE: usize = 32; // Maximum single write is 32 words
+const WORDS_PER_BANK: usize = 0x10000; // 64ki words per bank
+
+// Computes the flash Bank for the specified target location in words
+// from the beginning of flash.
+fn get_bank_from_target(target: usize) -> Option<Bank> {
+    if target < WORDS_PER_BANK {
+        Some(Bank::Zero)
+    } else if target < 2 * WORDS_PER_BANK {
+        Some(Bank::One)
+    } else {
+        None
+    }
+}
 
 impl<'d, A: Alarm<'d>, H: Hardware> super::flash::Flash<'d> for FlashImpl<'d, A, H> {
     fn erase(&self, page: usize) -> ReturnCode {
         if self.program_in_progress() { return ReturnCode::EBUSY; }
+        let target: usize = page * super::WORDS_PER_PAGE;
+
+        let maybe_bank = get_bank_from_target(target);
+        if maybe_bank.is_none() {
+            return ReturnCode::EINVAL;
+        }
+
+        self.write_bank.set(maybe_bank.unwrap());
+        self.write_bank_target.set(target % WORDS_PER_BANK);
         self.smart_program(ERASE_OPCODE, /*max_attempts*/ 45, /*final_pulse_needed*/ false,
-                           /*timeout_nanoseconds*/ 3_353_267,
-                           /*target*/ page * super::WORDS_PER_PAGE, /*size*/ 1);
+                           /*timeout_nanoseconds*/ 3_353_267, self.write_bank.get(),
```

Comment:
Where the _heck_ did this bizarre-looking magic number come from?

Reply:
I assume @jrvanwhy computed that from the H1B specs ...

Reply:
Maybe not yours to dig up, but IMO anything like this ought to include a comment with a pointer to the relevant spec.